core/src/test/java/kafka/log/remote/RemoteLogManagerTest.java

```diff
@@ -2437,11 +2437,10 @@ public void testRemoteLogSizeRetentionShouldFilterOutCopySegmentStartState()
         // s5. 11 segments with "COPY_SEGMENT_FINISHED" state. These are expected to be counted in when calculating remote storage log size
         //
         // Expected results (retention.size is 10240 (10 segments)):
-        // In the 1st run, the total remote storage size should be 1024 * 12 (s3, s5), so 2 segments (s1, s3) will be deleted
-        // due to retention size breached. s1 will be deleted even though it is not included in size calculation. But it's fine.
-        // The segment intended to be deleted will be deleted in the next run.
-        // In the 2nd run, the total remote storage size should be 1024 * 12 (s4, s5)
-        // so 2 segments (s4, s5[0]) will be deleted due to retention size breached.
+        // In the 1st run, the total remote storage size should be 1024 * 11 (s5), so 2 segments (s1, s3) will be deleted
+        // because they are dangling segments (s1 case should not happen as we always attempt a deletion when copy fails).
```

Comment:
Good point, @fvaleri ! After this PR, there should be 2 possible cases for `COPY_SEGMENT_STARTED` segment:
1. A segment is in `segmentIdsBeingCopied` set, and it's in `COPY_SEGMENT_STARTED` (or  `DELETE_SEGMENT_STARTED`/`DELETE_SEGMENT_FINISHED`) state. In this case, we won't delete it because of `segmentIdsBeingCopied`.
2. A segment is NOT in `segmentIdsBeingCopied` set, and it failed to copy to remote, and it'll either be in `DELETE_SEGMENT_STARTED` or `DELETE_SEGMENT_FINISHED` state. In this case, we'll retry `DELETE_SEGMENT_STARTED` deletion.

So that means, the S1 case (A segment is NOT in `segmentIdsBeingCopied` and in `COPY_SEGMENT_STARTED` state) is not practical anymore. Could you please update it? I'm thinking we can do 

1. simulate A segment is in `segmentIdsBeingCopied` set, and it's in `COPY_SEGMENT_STARTED` state. Or
2. keep it, because before this change, there will be dangling `COPY_SEGMENT_STARTED` segments, but we need to keep the comment clear, or
3. Remove S1?

What do you think?

Reply:
I went with 2 in order to keep trace of this discussion, which may be useful when debugging state updates.